spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8SecretsPropertySource.java

```diff
@@ -63,8 +62,11 @@ public Fabric8SecretsPropertySource(KubernetesClient client, String name, String
 				putDataFromSecret(secret, result, namespace);
 			}
 
-			client.secrets().inNamespace(namespace).withLabels(labels).list().getItems()
-					.forEach(s -> putDataFromSecret(s, result, namespace));
+			if (labels != null && !labels.isEmpty()) {
```

Comment:
while working on a diff PR, I had a test failure that kind of threw me off. After some debugging, I realized that fabric8 client is doing something not very correct, at least imo. [here is the explanation for it](https://github.com/spring-cloud/spring-cloud-kubernetes/pull/915/files#r748789603).

Basically, when doing something like: `client.secrets().inNamespace(namespace).withLabels(labels).list().getItems()` and passing an empty `labels` - this will match _all_ labels in that namespace. Without this fix, the unit test that I put in, fails. It fails because `secret-three` (that should have never been read anyway) overrides the value from `secret-two`. This is a defect. I am totally the one to be blamed for it, since I refactored the logic here in this manner. But boy, this was un-expected. I will raise a question in the fabric8 client also, I can't wrap my head around why this is the way it is, let's hope I miss something (I usually do)

Reply:
[here](https://github.com/fabric8io/kubernetes-client/issues/3590) is the fabric8 link to the question.

Reply:
Maybe add a comment to the code explaining the behavior and the check